ui/widgets/sortable.js

```diff
@@ -118,19 +118,19 @@ return $.widget( "ui.sortable", $.ui.mouse, {
 	},
 
 	_setHandleClassName: function() {
-		var that = this;
 		this._removeClass( this.element.find( ".ui-sortable-handle" ), "ui-sortable-handle" );
 		$.each( this.items, function() {
-			that._addClass(
-				this.instance.options.handle ?
-					this.item.find( this.instance.options.handle ) :
-					this.item,
-				"ui-sortable-handle"
-			);
+
+			// We use addClass() here, instead of _addClass(), because it is much faster.
+			( this.instance.options.handle ?
+				this.item.find( this.instance.options.handle ) :
+				this.item
+			).addClass( "ui-sortable-handle" );
 		} );
 	},
 
 	_destroy: function() {
+		this.element.find( ".ui-sortable-handle" ).removeClass( "ui-sortable-handle" );
```

Comment:
Looking at it, I'm not sure about this, it seems to be diverging from how classes are generally managed in jQuery UI.

@fnagel what do you think?

Reply:
I commented in the ticket that we use this same current workaround because users reported the same thing 4 years ago. The performance difference with 1000 items is 2 seconds instead of 2+ minutes in the browser

Reply:
@melloware Thanks, that's an important data point. I'd still want to hear from @fnagel who was involved with UI when the team was larger, I may be lacking some context from those times.

Reply:
Sorry for the late response. Still in the process of moving my office and flat.

The thing is that `_addClass` does keep track of the elements and does some stuff when the element is removed (afair). It's part of the `this.options.classes` functionality and I think its related to https://github.com/jquery/jquery-ui/pull/2037

One objection against this change is that we might reintroduce old issues we once fixed by adding the classes option. Seems we paid for bug fixes with performance penalty.

Reply:
Thanks for your reply. If you aren't comfortable with the patch, you can close it.

I have monkey patched that method in my project, so that I can still use v13 ;-)